3scale/porta
# Support OAS3 in Dev portal

> ⚠️ Merge only after #1594 

**What this PR does / why we need it**:

Updates `swagger-ui` to v3 in Dev portal in order to support OAS3 / Swagger 3. Also, `swagger-ui` is no longer added from the cdn but it is included as a webpack asset (node_modules).

Old versions of swagger will still be available for those customers who does not manually update.

Vanilla:
![Screen Shot 2020-01-28 at 11 19 32](https://user-images.githubusercontent.com/11672286/73255299-35c74700-41c0-11ea-97f6-8f692122a976.png)

With Scss patch (final verrsion):
<img width="1281" alt="Screen Shot 2020-01-31 at 17 34 30" src="https://user-images.githubusercontent.com/11672286/73556591-03cc1400-4450-11ea-8efa-1c02f39fb16f.png">


**Which issue(s) this PR fixes** 

[THREESCALE-2725: Start using SwaggerUI 3.18.3 or higher for OAS3 specs](https://issues.redhat.com/browse/THREESCALE-2725)

**Verification steps** 
1. Go to any Product's active docs section: apiconfig/services/2/api_docs/<id>/preview#/
2. Create a new spec (v3) i.e. [Swagger petstore](http://petstore.swagger.io:8080/api/v3/openapi.json)
3. Update the following templates at Dev Portal's Content section: /p/admin/cms/templates
    - Documentation
    ```liquid
   <h1>Documentation</h1>

   <p>Use our live documentation to learn about the Echo API</p>

    {% content_for javascripts %}
        {{ 'active_docs.js' | javascript_include_tag }}
    {% endcontent_for %}

    {% include 'shared/swagger_ui' %}
    ```
    - shared/swagger_ui
    ```rb
    -    SwaggerUIBundle({
    +    SwaggerUI({
           url: url,
    -      dom_id: 'swagger-ui-container',
    +      dom_id: '#swagger-ui-container',
           supportedSubmitMethods: ['get', 'post', 'put', 'delete', 'patch'],
           apisSorter: 'alpha', // can also be a function
           operationsSorter: 'method', // can also be 'alpha' or a function
    -      onComplete: function(swaggerApi, swaggerUi) {
    -        $(document.getElementById(swaggerUi.dom_id)).find('pre code').each(function(i, e {hljs.highlightBlock(e)})
    -      },
           docExpansion: 'list',
           transport: function (httpClient, obj) { ApiDocsProxy.execute(httpClient, obj) }
         });
    ```
4. At the dev portal > Documentation: it should correctly display the new (ugly) ui


**Special notes for your reviewer**:
Admin portal (active docs) support implemented in #1594 
Proper styling will be handled in a following PR (#??).
